tests/benchmark/test_bloatnet.py

```diff
@@ -0,0 +1,165 @@
+"""
+abstract: Tests that benchmarks EVMs to estimate the costs of  stateful opcodes.
+    Tests that benchmarks EVMs to estimate the costs of stateful opcodes..
+"""
+
+import pytest
+
+from ethereum_test_base_types import HashInt
+from ethereum_test_forks import Fork
+from ethereum_test_tools import (
+    Account,
+    Alloc,
+    Block,
+    BlockchainTestFiller,
+    Environment,
+    Storage,
+    Transaction,
+)
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+
+
+@pytest.mark.valid_from("Prague")
+@pytest.mark.parametrize("final_storage_value", [0x02 << 248, 0x02])
+def test_bloatnet(
+    blockchain_test: BlockchainTestFiller, pre: Alloc, fork: Fork, final_storage_value: int
+):
+    """
+    A test that calls a contract with many SSTOREs.
+
+    The first block will have many SSTORES that go from 0 -> 1
+    and the 2nd block will have many SSTORES that go from 1 -> 2
+    """
+    # Get gas costs for the current fork
+    gas_costs = fork.gas_costs()
+
+    # this is only used for computing the intinsic gas
+    data = final_storage_value.to_bytes(32, "big").rstrip(b"\x00")
+
+    storage = Storage()
+
+    # Initial gas for PUSH0 + CALLDATALOAD + POP (at the end)
+    totalgas = gas_costs.G_BASE * 2 + gas_costs.G_VERY_LOW
+    totalgas = totalgas + fork.transaction_intrinsic_cost_calculator()(calldata=data)
+    gas_increment = gas_costs.G_VERY_LOW * 2 + gas_costs.G_STORAGE_SET + gas_costs.G_COLD_SLOAD
+    sstore_code = Op.PUSH0 + Op.CALLDATALOAD
+    storage_slot: int = 0
+    while totalgas + gas_increment < Environment().gas_limit:
+        totalgas += gas_increment
+        sstore_code = sstore_code + Op.SSTORE(storage_slot, Op.DUP1)
+        storage[storage_slot] = final_storage_value
+        storage_slot += 1
+
+    sstore_code = sstore_code + Op.POP  # Drop last value on the stack
+
+    sender = pre.fund_eoa()
+    print(sender)
+    contract_address = pre.deploy_contract(
+        code=sstore_code,
+        storage=Storage(),
+    )
+
+    tx_0_1 = Transaction(
+        to=contract_address,
+        gas_limit=Environment().gas_limit,
+        data=(final_storage_value // 2).to_bytes(32, "big").rstrip(b"\x00"),
+        value=0,
+        sender=sender,
+    )
+    tx_1_2 = Transaction(
+        to=contract_address,
+        gas_limit=Environment().gas_limit,
+        data=final_storage_value.to_bytes(32, "big").rstrip(b"\x00"),
+        value=0,
+        sender=sender,
+    )
+
+    post = {contract_address: Account(storage=storage)}
+
+    blockchain_test(pre=pre, blocks=[Block(txs=[tx_0_1]), Block(txs=[tx_1_2])], post=post)
+
+
+# Warm reads are very cheap, which means you can really fill a block
+# with them. Only fill the block by a factor of SPEEDUP.
+SPEEDUP: int = 100
+
+
```

Comment:
We already have covered cold and warm SSTORE/SLOAD with same and different values here https://github.com/ethereum/execution-spec-tests/blob/main/tests/benchmark/test_worst_stateful_opcodes.py#L393

Are these tests different in some way?

Reply:
@gballet, the tests in `benchmark` are not specific to zkvms. (They are even used for PerfNet).

The test I linked are executing blocks where the full gas limit is used to do cold or warm reads, or to write slots to existent or non-existent storage slots. There's nothing specific to zkvms there, thus why I ask how these tests are different.

Reply:
I consider it has already been addressed by my first comment: it's not the same bytecode, not the same objectives (worst case vs perf), and not the same constraints. Ignacio's tests might not be specific to zkvms, but we do need something specific for us.